### ISSUE TITLE
🐛 BUG: {X} in body of markdown breaks build

### ISSUE THREAD

**Issue by stenehall** [snowpackjs/astro#721]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Having `{X}` in the body of markdown will crash the build

```---
title: 'More astro'
---

There's a simple secret to building a faster website — _just ship less_.

With this in the body {X}

Unfortunately, modern web development has been trending in the opposite direction—towards _more._ More JavaScript, more features, more moving parts, and ultimately more complexity needed to keep it all running smoothly.
```

### Steps to Reproduce

1. `npm init astro` using template <NAME>
2. Add markdown from description of bug
3. `yarn start`or `yarn build` 
5. Error! "ReferenceError: X is not defined"

**Comment by FredKSchott** [comment]
tbh this may be expected behavior, based on how our components-in-markdown support shakes out.

@natemoo-re is working on markdown + components now, assigning to him

**Comment by natemoo-re** [comment]
This is definitely a bug in the current implementation, since we _shouldn't_ throw on this. 

However, with Component support landing in `.md` files in #1406, this would be expected behavior.

**Comment by natemoo-re** [comment]
In `astro@next`, which implements components in markdown, this will be closed as wontfix.

This [reproduction](https://stackblitz.com/edit/github-3evjqr?file=src%2Fpages%2Findex.md)  shows the solution, which is to wrap braces like `{X}` in a string expression `{"{X}"}`.

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been addressed in [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

We still have a follow-up task to document this behavior before release.

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
